### ISSUE TITLE
[BUG] npm7 doesn't install packages that are only found on GitHub

### ISSUE THREAD

**Issue by vlilloh** [npm/cli#1959]
### Current Behavior:

It doesn't install packages that are only found on GitHub.

### Expected Behavior:

It should install packages that are only found on GitHub.

### Steps To Reproduce:

1. With this config in `package.json`, e.g.: `"dependencies": { "@gastromatic/react-dates": "1.0.7" }`
2. With this config in `.npmrc`, e.g.: `@gastromatic:registry=https://npm.pkg.github.com`
3. GitHub Personal Access Token with `repo` and `read:packages` scopes.
4. Run `npx npm@7.0.0 -reg https://npm.pkg.github.com` and login.
5. Run `npx npm@7.0.0 i`
6. See error:
```
npm ERR! code E401
npm ERR! Incorrect or missing password.
npm ERR! If you were trying to login, change your password, create an
npm ERR! authentication token or enable two-factor authentication then
npm ERR! that means you likely typed your password in incorrectly.
npm ERR! Please try again, or recover your password at:
npm ERR!     https://www.npmjs.com/forgot
npm ERR! If you were doing some other operation then your saved credentials are
npm ERR! probably out of date. To correct this please try logging in again with:
npm ERR!     npm login
npm ERR! A complete log of this run can be found in:
npm ERR!     ***
```

With these steps npm6 does install packages that are only found on GitHub.

### Environment:

- OS: MacOS 10.15.7
- Node: 14.13.1
- npm: 7.0.0

**Comment by nlf** [comment]
i believe we have resolved this auth bug, can you try updating to 7.0.1 `npm i -g npm@next-7` and see if it takes care of the issue?

**Comment by vlilloh** [comment]
npm v7.0.1 takes care of the issue. Thank you.
